srcCpp/RTIDDSImpl.cxx

```diff
@@ -1134,7 +1130,8 @@ class RTIDynamicDataPublisher : public IMessagingWriter
 
     bool waitForPingResponse() {
         if (_pongSemaphore != NULL) {
-            if (!RTIOsapiSemaphore_take(_pongSemaphore, NULL)) {
+            if (RTIOsapiSemaphore_take(_pongSemaphore, NULL)
```

Comment:
I would distinguish between RTI_OSAPI_SEMAPHORE_STATUS_TIMEOUT and RTI_OSAPI_SEMAPHORE_STATUS_ERROR

    bool waitForPingResponse(int timeout) 
    {
        RTIOsapiSemaphoreStatus semStatus;
        struct RTINtpTime blockDurationIn;
        RTINtpTime_packFromMillisec(blockDurationIn, 0, timeout);

        if(_pongSemaphore != NULL)
        {
            semStatus = RTIOsapiSemaphore_take(
                    _pongSemaphore,
                    &blockDurationIn);

            if(semStatus != RTI_OSAPI_SEMAPHORE_STATUS_OK)
            {
                if (semStatus == RTI_OSAPI_SEMAPHORE_STATUS_TIMEOUT) {
                    fprintf(stderr,"Timeout error taking semaphore\n");
                } else {
                    fprintf(stderr,"Unexpected error taking semaphore\n");
                }
                return false;
            }
        }
        return true;
    }

Reply:
@javmorcas Sounds good to me